toqito/state_opt/state_exclusion.py

```diff
@@ -45,15 +49,44 @@ def state_exclusion(
         .. math::
             \begin{equation}
                 \begin{aligned}
-                    \text{maximize:} \quad & \text{Tr}(Y)
+                    \text{maximize:} \quad & \text{Tr}(Y) \\
                     \text{subject to:} \quad & Y \leq M_1, \\
                                              & Y \leq M_2, \\
                                              & \vdots \\
                                              & Y \leq M_n, \\
-                                             & Y \text{Herm}(\mathcal{X}).
+                                             & Y \in\text{Herm}(\mathcal{X}).
                 \end{aligned}
             \end{equation}
 
+    For :code:`strategy = "unambiguous"`, Bob never provides an incorrect answer, although it is
+    possible that his answer is inconclusive. This function then yields the probability of an inconclusive outcome.
+
+    In that case, this function implements the following semidefinite program that provides the
+    optimal probability with which Bob can conduct unambiguous quantum state distinguishability.
+
+    .. math::
+        \begin{align*}
+            \text{minimize:} \quad & \text{Tr}\left(
+                \left(\sum_{i=1}^n p_i\rho_i\right)\left(\mathbb{I}-\sum_{i=1}^nM_i\right)
+                \right) \\
+            \text{subject to:} \quad & \sum_{i=1}^nM_i \geq \mathbb{I},\\
+                                     & M_1, \ldots, M_n \geq 0, \\
+                                     & \langle M_1, \rho_1 \rangle, \ldots, \langle M_n, \rho_n \rangle =0
+        \end{align*}
+
+    .. math::
+        \begin{align*}
+            \text{maximize:} \quad & 1 - \text{Tr}(N) \\
+            \text{subject to:} \quad & a_1p_1\rho_1, \ldots, a_np_n\rho_n \geq \sum_{i=1}^np_i\rho_i - N,\\
+                                     & N \geq 0,\\
+                                     & a_1, \ldots, a_n \in\mathbb{R}
+        \end{align*}
+
+
+    .. warning::
+        Note that it only makes sense to exclude unambiguously when the pure states are linearly
```

Comment:
Fundamentally, I'm not even sure it should be a warning: it just happens that we can always succeed with probability $1$ in that case and that we know a closed-form expression. This could be transparent to the user if we directly set the variables to what they should be.

For the dual it's easy, we can take $N=\sum\limits_ip_i\rho_i$ and $a_i=1$.

For the primal however, I'm not sure (there's probably a closed-form expression, but I don't know it!)

Reply:
True, it probably doesn't need to be a warning--it's more so a property of the fact that linearly dependent quantum states will always yield a value of 1. Perhaps removing the warning and putting this info in the docstring would be okay?

Aside from that though, I'm happy with the PR and can merge whenever you feel it to be ready, @tnemoz !